server/market/orderrouter.go

```diff
@@ -452,6 +475,46 @@ func (r *OrderRouter) extractMarket(prefix *msgjson.Prefix) (MarketTunnel, *msgj
 	return tunnel, nil
 }
 
+// SuspendEpoch holds the index and end time of final epoch marking the
+// suspension of a market.
+type SuspendEpoch struct {
+	Idx int64
+	End time.Time
+}
+
+// SuspendMarket schedules a suspension of a given market, with the option to
+// persist the orders on the book (or purge the book automatically on market
+// shutdown). The scheduled final epoch and suspend time are returned.
+// TODO: track running, suspended, and scheduled-suspended markets.
+func (r *OrderRouter) SuspendMarket(mktName string, asSoonAs time.Time, persistBooks bool) *SuspendEpoch {
+	mkt, found := r.tunnels[mktName]
+	if !found {
+		return nil
+	}
+
+	idx, t := mkt.Suspend(asSoonAs, persistBooks)
+	return &SuspendEpoch{
+		Idx: idx,
+		End: t,
+	}
+}
+
+func (r *OrderRouter) Suspend(asSoonAs time.Time, persistBooks bool) map[string]*SuspendEpoch {
+
+	suspendTimes := make(map[string]*SuspendEpoch, len(r.tunnels))
+	for name, mkt := range r.tunnels {
+		idx, ts := mkt.Suspend(asSoonAs, persistBooks)
+		suspendTimes[name] = &SuspendEpoch{Idx: idx, End: ts}
+	}
```

Comment:
Probably remove, or comment if this is the approach to take for a more graceful full shutdown.